src/blockdata/constants.rs

```diff
@@ -50,6 +50,8 @@ pub const MIN_TRANSACTION_WEIGHT: u32 = 4 * 60;
 pub const WITNESS_SCALE_FACTOR: usize = 4;
 /// The maximum allowed number of signature check operations in a block
 pub const MAX_BLOCK_SIGOPS_COST: i64 = 80_000;
+/// The maximum script length in bytes.
+pub const MAX_SCRIPT_SIZE: usize = 10_000;
```

Comment:
I know this is how it is written in bitcoin core, but this is not true post tapscript. Scripts can exceed `MAX_SCRIPT_SIZE`. I wanted PR to core to change this but never got around to doing this. 

So, someone looking at the code might think that tapscript witness scripts are restricted by 10_000 which is not true. 

I think what we need is `MAX_SCRIPTPUBKEY_SIZE` here?

Reply:
I think this dust check applies to the scriptPubKey only and as such tapscript bigger than 10_000 bytes must still comply to the dust evaluation ?

In fact, reading again core, any scriptpubkey template bigger than 10_000 bytes won't be standard, including op_return output because of `nMaxDatacarrierBytes`. So I guess this PR can be closed and core could be updated to change `IsUnspendable` in `GetDustThreshold` to `IsOpReturn` to remove the confusion ?